javascript/ql/lib/semmle/javascript/frameworks/Express.qll

```diff
@@ -1071,4 +1072,23 @@ module Express {
 
     override predicate definitelyResumesDispatch() { none() }
   }
+
+  /**
+   * An express route setup configured with the `cors` package.
+   */
+  class CorsConfiguration extends DataFlow::MethodCallNode {
+    CorsConfiguration() { exists(Express::RouteSetup setup | this = setup | setup.isUseCall()) }
```

Comment:
This characteristic predicate holds for any `.use()` call, you haven't restricted it to only be the ´.use() ´ calls that are configured with the Cors package.  

I think you should add `Cors::Cors` as a field on this class. That should make it easier to implement the restriction. 

Additionally (as I have mentioned before). 
The cors package can also be used like: app.get('/products/:id', cors(corsOptions) ....
With RouteSetup, I think it should be easy to model that.  
So if `setup.isUseCall()`, then the first argument should be the call to Cors.  
And if `setup.isUseCall()` doesn't hold, then the later arguments can be the call to Cors.

Reply:
Could it be simplified to just `setup.getAnArgument() instanceof Cors::Cors`?

Reply:
Not quite. 
Your `Cors::Cors getArgument()` predicate on the `CorsConfiguration` class doesn't work, because that predicate just assumes that the first argument is the `cors` configuration.  

I've added some suggestions to use a field such that the result of `getArgument` depends on the routesetup in the characteristic predicate.